pageserver/src/page_service.rs

```diff
@@ -218,9 +218,23 @@ async fn page_service_conn_main(
     // no write timeout is used, because the kernel is assumed to error writes after some time.
     let mut socket = tokio_io_timeout::TimeoutReader::new(socket);
 
+    let default_timeout_ms = 10 * 1000;
+    let socket_timeout_ms = (|| {
+        fail::fail_point!("simulated-bad-compute-connection", |avg_timeout_ms| {
+            if let Some(avg_timeout_ms) = avg_timeout_ms {
+                let avg = avg_timeout_ms.parse::<i64>().unwrap() as f32;
+                let u = rand::random::<f32>();
+                ((1.0 - u).ln() / (-avg)) as u64
+            } else {
+                default_timeout_ms
+            }
+        });
+        default_timeout_ms
+    })();
+
     // timeout should be lower, but trying out multiple days for
```

Comment:
Please can you remove this comment "timeout should be lower..." before merging -- it's no longer relevant now that the default timeout is reduced to 4205